.github/workflows/build_and_test.yml

```diff
@@ -938,7 +938,7 @@ jobs:
       HIVE_PROFILE: hive2.3
       GITHUB_PREV_SHA: ${{ github.event.before }}
       SPARK_LOCAL_IP: localhost
-      ORACLE_DOCKER_IMAGE_NAME: gvenzl/oracle-xe:21.3.0
+      ORACLE_DOCKER_IMAGE_NAME: gvenzl/oracle-free:23.3
```

Comment:
Should we backport this patch to branch-3.x? Otherwise, their daily tests will use `gvenzl/oracle-free:23.3`, but the test code in branch-3.x still uses `gvenzl/oracle-xe:21.3.0`.

It seems that the `OracleIntegrationSuite` in the `Docker integration tests` task of the daily tests for branch-3.x has failed recently.

- branch-3.3: https://github.com/apache/spark/actions/runs/6609791712/job/17950549755
- branch-3.4: https://github.com/apache/spark/actions/runs/6611049884/job/17954225189
- branch-3.5: https://github.com/apache/spark/actions/runs/6612344747/job/17958021656

Reply:
Or can we use a solution similar to https://github.com/apache/spark/pull/43306?

For example, for branch-3.x, explicitly add the environment variable `ORACLE_DOCKER_IMAGE_NAME` like

```
envs: >-
        {
          "SCALA_PROFILE": "scala2.13",
          "PYTHON_TO_TEST": "",
          "ORACLE_DOCKER_IMAGE_NAME": "gvenzl/oracle-xe:21.3.0"
        }
```

Reply:
Backporting would be better for all downstream, @LuciferYang . Feel free to backport.

Reply:
> Or can we use a solution similar to #43306?
> 
> For example, for branch-3.x, explicitly add the environment variable `ORACLE_DOCKER_IMAGE_NAME` like
> 
> ```
> envs: >-
>         {
>           "SCALA_PROFILE": "scala2.13",
>           "PYTHON_TO_TEST": "",
>           "ORACLE_DOCKER_IMAGE_NAME": "gvenzl/oracle-xe:21.3.0"
>         }
> ```

Hmm... Are we sure that this method can't override this environment variable? If possible, this solution would be relatively simple ... @HyukjinKwon

Reply:
> Or can we use a solution similar to #43306?
> 
> For example, for branch-3.x, explicitly add the environment variable `ORACLE_DOCKER_IMAGE_NAME` like
> 
> ```
> envs: >-
>         {
>           "SCALA_PROFILE": "scala2.13",
>           "PYTHON_TO_TEST": "",
>           "ORACLE_DOCKER_IMAGE_NAME": "gvenzl/oracle-xe:21.3.0"
>         }
> ```

Let me see if I can test this in my local repository.